### PR TITLE
Implement DAS readiness and liveness probes

### PR DESCRIPTION
**Pull request by a-thomas-22**
Implements readiness and liveness checks for the DAS.

If the rpc is enabled, that is used for liveness and readiness. If only the rest service is enabled, that is used for liveness and readiness.

Adds a test case for a DAS mirror configuration, and converts the current test case for a DAC member configuration.

### CODE REVIEW

File: charts/das/templates/_helpers.tpl [hunk 1]
```diff
@@ -62,13 +62,22 @@ Create the name of the service account to use
 {{- end }}
 
 {{- define "startupProbe" -}}
-curl "http://localhost:{{ .Values.configmap.data.http.port }}{{ .Values.configmap.data.http.rpcprefix }}" -H "Content-Type: application/json" \
-         -sd "{\"jsonrpc\":\"2.0\",\"id\":0,\"method\":\"eth_syncing\",\"params\":[]}" \
-         | jq -ne "input.result == false"
+curl "http://localhost:{{ index .Values.configmap.data "rpc-port" }}" -X POST \
```
Comment (gmas) [inline charts/das/templates/_helpers.tpl:65]: Looks like`.Values.configmap.data "rpc-port"` is missing from the values YAML

File: charts/das/templates/statefulset.yaml [hunk 2]
```diff
@@ -101,29 +101,69 @@ spec:
...
+            initialDelaySeconds: {{ .initialDelaySeconds }}
+            periodSeconds: {{ .periodSeconds }}
+            timeoutSeconds: {{ .timeoutSeconds }}
+            failureThreshold: {{ .failureThreshold }}
+            successThreshold: {{ .successThreshold }}
+            {{- end }}
+            {{- if and (not (index .Values.configmap.data "enable-rpc")) (index .Values.configmap.data "enable-rest") }}
+            httpGet:
+              path: /health
+              port: {{ index .Values.configmap.data "rest-port" }}
```
Comment (gmas) [inline charts/das/templates/statefulset.yaml:114]: same as for `.Values.configmap.data "rpc-port"` can't find a reference for `.Values.configmap.data "rest-port"`

File: charts/das/templates/statefulset.yaml [hunk 3]
```diff
@@ -101,29 +101,69 @@ spec:
...
```
Comment (gmas) [inline charts/das/templates/statefulset.yaml:136]: `.Values.configmap.data "rest-port"` not defined

File: charts/das/templates/statefulset.yaml [hunk 4]
```diff
@@ -101,29 +101,69 @@ spec:
...
```
Comment (gmas) [inline charts/das/templates/statefulset.yaml:158]: `.Values.configmap.data "rest-port"` not defined

File: charts/das/templates/_helpers.tpl [hunk 5]
```diff
@@ -62,13 +62,22 @@ Create the name of the service account to use
...
+      -H 'Content-Type: application/json' \
+      -d '{"jsonrpc":"2.0","id":0,"method":"das_healthCheck","params":[]}'
 {{- end -}}
 
+{{- define "livenessProbe" -}}
+curl "http://localhost:{{ index .Values.configmap.data "rpc-port" }}" -X POST \
```
Comment (gmas) [inline charts/das/templates/_helpers.tpl:71]: `.Values.configmap.data "rpc-port"` is not defined

File: charts/das/templates/_helpers.tpl [hunk 6]
```diff
@@ -62,13 +62,22 @@ Create the name of the service account to use
...
+      -H 'Content-Type: application/json' \
+      -d '{"jsonrpc":"2.0","id":0,"method":"das_healthCheck","params":[]}'   
+{{- end -}}
 
-
+{{- define "readinessProbe" -}}
+curl "http://localhost:{{ index .Values.configmap.data "rpc-port" }}" -X POST \
```
Comment (gmas) [inline charts/das/templates/_helpers.tpl:77]: `.Values.configmap.data "rpc-port"` is not defined

**Review by gmas — COMMENTED** [review]
IIUC the values for "rpc-port" and "rest-port" are not set

File: charts/das/templates/statefulset.yaml (hunk 2 above)
Comment (a-thomas-22) [reply charts/das/templates/statefulset.yaml:114]: https://github.com/OffchainLabs/community-helm-charts/blob/5005b0d952b773936ea0085d6e7290fe94326f7d/charts/das/values.yaml#L147

File: charts/das/templates/_helpers.tpl (hunk 1 above)
Comment (a-thomas-22) [reply charts/das/templates/_helpers.tpl:65]: https://github.com/OffchainLabs/community-helm-charts/blob/5005b0d952b773936ea0085d6e7290fe94326f7d/charts/das/values.yaml#L153

Comment (a-thomas-22) [reply charts/das/templates/_helpers.tpl:65]: That is intentional. That values file is a test case for a das mirror which would not have the rpc enabled.

File: charts/das/templates/statefulset.yaml (hunk 3 above)
Comment (a-thomas-22) [reply charts/das/templates/statefulset.yaml:136]: https://github.com/OffchainLabs/community-helm-charts/blob/5005b0d952b773936ea0085d6e7290fe94326f7d/charts/das/values.yaml#L147

**Comment by a-thomas-22** [comment]
> IIUC the values for "rpc-port" and "rest-port" are not set

They are set in the main values file.

File: charts/das/templates/_helpers.tpl (hunk 5 above)
Comment (a-thomas-22) [reply charts/das/templates/_helpers.tpl:71]: same as above

File: charts/das/templates/_helpers.tpl (hunk 6 above)
Comment (a-thomas-22) [reply charts/das/templates/_helpers.tpl:77]: same as above

**Review by gmas — APPROVED** [review]

**Pull request merged**